tests/serializers/test_data.py

```diff
@@ -420,70 +406,97 @@ def inherited_compare(testcase, pk, klass, data):
 ]
 
 
-# Because Oracle treats the empty string as NULL, Oracle is expected to fail
-# when field.empty_strings_allowed is True and the value is None; skip these
-# tests.
-if connection.features.interprets_empty_strings_as_nulls:
-    test_data = [
-        data
-        for data in test_data
-        if not (
-            data[0] == data_obj
-            and data[2]._meta.get_field("data").empty_strings_allowed
-            and data[3] is None
-        )
-    ]
-
-
-if not connection.features.supports_index_on_text_field:
-    test_data = [data for data in test_data if data[2] != TextPKData]
-
-
 class SerializerDataTests(TestCase):
     pass
 
 
-def serializerTest(self, format):
-    # FK to an object with PK of 0. This won't work on MySQL without the
-    # NO_AUTO_VALUE_ON_ZERO SQL mode since it won't let you create an object
-    # with an autoincrement primary key of 0.
-    if connection.features.allows_auto_pk_0:
-        test_data.extend(
-            [
-                (data_obj, 0, Anchor, "Anchor 0"),
-                (fk_obj, 465, FKData, 0),
-            ]
-        )
-
-    # Create all the objects defined in the test data
+def assert_serializer(self, format, data):
+    # Create all the objects defined in the test data.
     objects = []
-    instance_count = {}
-    for func, pk, klass, datum in test_data:
+    for manager, pk, model, data_value in data:
         with connection.constraint_checks_disabled():
-            objects.extend(func[0](pk, klass, datum))
+            objects.extend(manager.create_object(pk, model, data_value))
 
-    # Get a count of the number of objects created for each class
-    for klass in instance_count:
-        instance_count[klass] = klass.objects.count()
+    # Get a count of the number of objects created for each model class.
+    instance_count = {}
```

Comment:
Nitpick: if we're refactoring anyway, I wonder if `instance_counts` would be a slightly more accurate name.